plugin/trino-iceberg/src/main/java/io/trino/plugin/iceberg/AllFilesTable.java

```diff
@@ -0,0 +1,69 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.trino.plugin.iceberg;
+
+import io.trino.plugin.iceberg.util.PageListBuilder;
+import io.trino.spi.Page;
+import io.trino.spi.connector.ConnectorPageSource;
+import io.trino.spi.connector.ConnectorSession;
+import io.trino.spi.connector.ConnectorTransactionHandle;
+import io.trino.spi.connector.FixedPageSource;
+import io.trino.spi.connector.SchemaTableName;
+import io.trino.spi.predicate.TupleDomain;
+import io.trino.spi.type.TypeManager;
+import org.apache.iceberg.DataTask;
+import org.apache.iceberg.MetadataTableType;
+import org.apache.iceberg.Table;
+import org.apache.iceberg.TableScan;
+import org.apache.iceberg.types.Type;
+
+import java.util.List;
+import java.util.Map;
+
+import static org.apache.iceberg.MetadataTableUtils.createMetadataTableInstance;
+
+public class AllFilesTable
+        extends AbstractFilesTable
+{
+    public AllFilesTable(SchemaTableName tableName, TypeManager typeManager, Table icebergTable)
+    {
+        super(tableName, typeManager, icebergTable);
+    }
+
+    @Override
+    public ConnectorPageSource pageSource(ConnectorTransactionHandle transactionHandle, ConnectorSession session, TupleDomain<Integer> constraint)
+    {
+        return new FixedPageSource(buildPages(getIcebergTable()));
+    }
+
+    protected List<Page> buildPages(Table icebergTable)
+    {
+        PageListBuilder pagesBuilder = PageListBuilder.forTable(getTableMetadata());
+        Map<Integer, Type> idToTypeMapping = getIcebergIdToTypeMapping(icebergTable.schema());
+        Table filesTable = createMetadataTableInstance(icebergTable, MetadataTableType.ALL_DATA_FILES);
```

Comment:
@findinpath please see.

using Iceberg Metadata APIs/Table to scan the files.

this will help to use the changes/fixes done in the Iceberg APIs.
Also, the optimization like de-duping manifests files in case of `all_files` is done in Iceberg APIs' which Trino can use automatically. 

as adding manifests to a set at
https://github.com/apache/iceberg/blob/718fc5ae9651d016dcfd2cb80d57bfb7366e984b/core/src/main/java/org/apache/iceberg/BaseAllMetadataTableScan.java#L69

used by AllDataFiles at
https://github.com/apache/iceberg/blob/718fc5ae9651d016dcfd2cb80d57bfb7366e984b/core/src/main/java/org/apache/iceberg/AllDataFilesTable.java#L69

We can also refactor other metadata tables to use Iceberg APIs.